foxpuppet/windows/browser/tab_bar.py

```diff
@@ -0,0 +1,98 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/
+"""Creates tab_bar object to interact with the Firefox tabs and tabbar."""
+
+from selenium.webdriver.common.by import By
+
+from foxpuppet.region import Region
+
+
+class Tab_bar(Region):
+    """Representation of the tab bar which contains the tabs.
+
+    Args:
+        window (:py:class:`BaseWindow`): Window object this region appears in.
+        root
+            (:py:class:`~selenium.webdriver.remote.webelement.WebElement`):
+            WebDriver element object that serves as the root for the
+            region.
+
+    """
+
+    _new_tab_button_locator = (By.ID, 'new-tab-button')
+    _tabs_locator = (By.TAG_NAME, 'tab')
+
+    @property
+    def tabs(self):
+        """Return a list of tabs.
+
+        Returns: :py:class:`~foxpuppet.window.browser.tab_bar.Tab`
+
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            tabs = [self.Tab(self, el) for el in
+                    self.selenium.find_elements(*self._tabs_locator)]
+        # Assign handles
+        for tab, handle in zip(tabs, self.selenium.window_handles):
+            tab.handle = handle
+        return tabs
+
+    def open_new_tab(self):
+        """Open a new tab in the current window.
+
+        Returns: list of :py:class:`Tab`.
+
+        """
+        current_tabs = len(self.selenium.window_handles)
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*self._new_tab_button_locator).click()
+        self.wait.until(
+            lambda _: len(self.selenium.window_handles) != current_tabs)
+        return self.tabs
```

Comment:
Why return the list of tabs? It would make more sense to return the new tab so we could do something like:

```python
new_tab = tab_bar.open_new_tab()
new_tab.close()
```

Reply:
It would all make more sense if code or the technique which we have already written for Firefox Puppet would be taken. It said this already a couple of times, and here is one more example. It would help us to not have such a lot of review cycles.

Reply:
Currently in Firefox Puppeteer: 
```
Wait(self.marionette).until(
            lambda mn: len(mn.window_handles) == len(start_handles) + 1,
            message='No new tab has been opened.')

        handles = self.marionette.window_handles
        [new_handle] = list(set(handles) - set(start_handles))
        [new_tab] = [tab for tab in self.tabs if tab.handle == new_handle]

        # if the new tab is the currently selected tab, switch to it
        if new_tab == self.selected_tab:
            new_tab.switch_to()
```

Which is what Dave is suggesting anyways. I will model this a little closer.